### ISSUE TITLE
Investigate probable bug in the Spanner integration tests

### ISSUE THREAD

**Issue by Rishabh-V** [googleapis/google-cloud-dotnet#9204]
Please follow the below thread for context.        

 I think this should be, it has probably always worked because where we run with the emulator the environment variable with the DB name is not set, but still, this is a bug.
```suggestion
        if (Fresh || SpannerClientCreationOptions.UsesEmulator)
```

_Originally posted by @amanda-tarafa in https://github.com/googleapis/google-cloud-dotnet/pull/9164#discussion_r986921958_

**Comment by Rishabh-V** [comment]
Hi @amanda-tarafa, I have created a placeholder for the issue. Please update/edit with more details. Thank you.

**Comment by Rishabh-V** [comment]
Hi @amanda-tarafa, I have created a placeholder for the issue. Please update/edit with more details. Thank you.

**Comment by jskeet** [comment]
Have set the type as "cleanup" as we don't believe it's a bug in the *production* code.

**Comment by amanda-tarafa** [comment]
The main issue in the current implementation is that we assume the db is fresh if and only if the environment variable is not set. But reality is:

- If the environment variable is not set, the database is always fresh.
- If the environment variale is set, the database might not be fresh.
  - In particular, when running on the emulator, we always attempt to create the instance, and ignore the AlreadyExists error. But if we succeeded in creating the instance, then the DB will be fresh even if the env var is present.

Proposal: Always attempt to create the DB, and mark fresh if successfull.

Note: this is extremely low priority, it only affects test code and it doesn't affect any of our CI environments.

**Comment by anuragsrivstv** [comment]
Issue fixed in #10044
